File: Python/4.median-of-two-sorted-arrays.py
#
# @lc app=leetcode id=4 lang=python3
#
# [4] Median of Two Sorted Arrays
#

# @lc code=start
from typing import List

class Solution:
    def findMedianSortedArrays(self, nums1: List[int], nums2: List[int]) -> float:
        ov = sorted(nums1 + nums2)
        l = len(ov)
        if len(ov) % 2 == 0:
            return (ov[l // 2] + ov[l // 2 - 1]) / 2
        return ov[l // 2]
# @lc code=end

File: C++/4.median-of-two-sorted-arrays.cpp
/*
 * @lc app=leetcode id=4 lang=cpp
 *
 * [4] Median of Two Sorted Arrays
 */

// @lc code=start
#include <vector>
#include <algorithm>

using namespace std;

class Solution
{
public:
    double findMedianSortedArrays(vector<int> &nums1, vector<int> &nums2)
    {
        vector<int> merged(nums1.size() + nums2.size());
        merge(nums1.begin(), nums1.end(), nums2.begin(), nums2.end(), merged.begin());
        sort(merged.begin(), merged.end());

        size_t len = merged.size();
        if (len % 2 == 0)
        {
            return ((double)merged[len / 2] + (double)merged[len / 2 - 1]) / 2;
        }
        else
        {
            return merged[len / 2];
        }
    }
};
// @lc code=end
